documentation/asciidoc/computers/linux_kernel/building.adoc

```diff
@@ -1,342 +1,462 @@
 ----
-sudo env PATH=$PATH make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- INSTALL_MOD_PATH=mnt/ext4 modules_install
+$ sudo env PATH=$PATH make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- INSTALL_MOD_PATH=mnt/root modules_install
 ----
 
-Finally, copy the kernel and Device Tree blobs onto the SD card, making sure to back up your old kernel:
-
-===== For 32-bit
+Next, build and install the kernel, modules, and Device Tree blobs into the boot partition, backing up your original kernel.
 
-[,bash]
+* To build and install the 64-bit kernel image, overlays, README, and unmount the partitions, run the following commands:
++
+[source,console]
 ----
-sudo cp mnt/fat32/$KERNEL.img mnt/fat32/$KERNEL-backup.img
-sudo cp arch/arm/boot/zImage mnt/fat32/$KERNEL.img
-# Choose one of the following based on the kernel version
-  # For kernels up to 6.4:
-  sudo cp arch/arm/boot/dts/*.dtb mnt/fat32/
-  # For kernel 6.5 and above:
-  sudo cp arch/arm/boot/dts/broadcom/*.dtb mnt/fat32/
-sudo cp arch/arm/boot/dts/overlays/*.dtb* mnt/fat32/overlays/
-sudo cp arch/arm/boot/dts/overlays/README mnt/fat32/overlays/
-sudo umount mnt/fat32
-sudo umount mnt/ext4
+$ sudo cp mnt/boot/$KERNEL.img mnt/boot/$KERNEL-backup.img
+$ sudo cp arch/arm64/boot/Image mnt/boot/$KERNEL.img
+$ sudo cp arch/arm64/boot/dts/broadcom/*.dtb mnt/boot/
+$ sudo cp arch/arm64/boot/dts/overlays/*.dtb* mnt/boot/overlays/
+$ sudo cp arch/arm64/boot/dts/overlays/README mnt/boot/overlays/
+$ sudo umount mnt/boot
+$ sudo umount mnt/root
 ----
 
-===== For 64-bit
+* To build the 32-bit kernel:
++
+. Run the following commands to build the modules and Device tree blobs:
++
+[source,console]
+----
+$ sudo cp mnt/boot/$KERNEL.img mnt/boot/$KERNEL-backup.img
+$ sudo cp arch/arm/boot/zImage mnt/boot/$KERNEL.img
+----
 
-[,bash]
+. Depending on your kernel version, run the following command:
+  * For kernels up to version 6.4:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/*.dtb mnt/boot/
+----
+* For kernels version 6.5 and above:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/broadcom/*.dtb mnt/boot/
 ----
-sudo cp mnt/fat32/$KERNEL.img mnt/fat32/$KERNEL-backup.img
-sudo cp arch/arm64/boot/Image mnt/fat32/$KERNEL.img
-sudo cp arch/arm64/boot/dts/broadcom/*.dtb mnt/fat32/
-sudo cp arch/arm64/boot/dts/overlays/*.dtb* mnt/fat32/overlays/
-sudo cp arch/arm64/boot/dts/overlays/README mnt/fat32/overlays/
-sudo umount mnt/fat32
-sudo umount mnt/ext4
+. Finally, copy over the overlays, README, and image, and unmount the partitions:
++
+[source,console]
 ----
+$ sudo cp arch/arm/boot/dts/overlays/*.dtb* mnt/boot/overlays/
+$ sudo cp arch/arm/boot/dts/overlays/README mnt/boot/overlays/
+$ sudo umount mnt/boot
+$ sudo umount mnt/root
+----
+
+TIP: On multi-core Raspberry Pi models, the `-j <n>` option distributes work between cores. This can speed up compilation significantly. Run `nproc` to see how many processors you have; we recommend passing a number 1.5x your number of processors.
```

Comment:
This is the cross-compile section, so no point talking about "multi-core Raspberry Pi models" :stuck_out_tongue_winking_eye: 
But we **do** still want to recommend the `-j` option, as a multi-core desktop PC is even more likely than a multi-core Raspberry Pi. (so should probably also demonstrate that in the commands above where we `make` the `Image` or `zImage` files)

Reply:
Great point! Moved this tip up closer to the `make` step, and added `-j12` because 8 cores seems like a reasonable expectation for modern desktop PCs.